arch/common/timing.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
```

Comment:
Getting deja vu reading this code.
Then I realized we already have it in arch/arm.
Wouldn't it be better to just keep this file, and implemnnent any differences in the arch layer?

Reply:
The idea is that this file is for architectures and SoCs without high resolution timers. So just get the values from the default timer driver. Arch code can override this, as in ARM (SysTick) and x86 (TSC). SoC code can also override these for their specific timers, e.g. NRF and MEC1501. This layering is a bit tricky. I am not sure if `__weak` alone can support this.

Reply:
I might be nitpicking as on ARC default Timers (either ARC Timer0/1 or GFRC in SMP system) is synchronized with CPU clock which makes those timers pretty high-resolution (as least for measurements sake as we don't care about latency of 0.5 a cycle I guess). Or is there any benefit of having architecture- or SoC-specific implementation of these functions other than higher precision of the timer being used?

Reply:
@abrodkin  we will need an implementation of those functions for ARC based on the high res timer. Is this the same thing as in arch/arc/core/timestamp.c?

Reply:
Both arch/arc/core/timestamp.c and the timer driver arcv2_timer0.c use the timer0 that is part of the ARC core and runs at CPU clock frequency. Timestamp.c doesn't seem to take into account timer setting and wrapping in the timer driver, that is likely a problem.
Depending on ARC configuration, there can be a second timer, and 2 separate secure mode timers. There can be a counter as well, and for SMP a global counter.
Problem is that not all of these have to be present in in all ARC instances, ARC users choose a configuration. So we try to work with the minimum, though using more timers/counters would make implementations simpler.